MdePkg/Library/BaseLib/RiscV64/Unaligned.c

```diff
@@ -0,0 +1,281 @@
+/** @file
+  Unaligned access functions of BaseLib for RISCV64.
+
+  Copyright (c) 2025, Xiang W <wangxiang@iscas.ac.cn>. All rights reserved.<BR>
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+
+**/
+#include <Library/BaseLib.h>
+#include "BaseLibInternals.h"
+
+/**
+  Reads a value from memory that may be unaligned.
+
+  This function returns the value pointed to by Buffer. The function
+  guarantees that the read operation does not produce an alignment fault.
+
+  If the Buffer is NULL, then ASSERT().
+
+  @param  Buffer   The pointer to a value that may be unaligned.
+  @param  BytesLen Byte length of the value to be read
+
+  @return The value read from Buffer.
+
+**/
+inline
+STATIC
+UINT64
+ReadUnalignedWithSize (
+  IN CONST VOID  *Buffer,
+  IN UINT64      BytesLen
+  )
+{
+  UINT64  Index = 0;
+  UINT64  Value = 0;
+
+  ASSERT (Buffer != NULL);
+  if (RiscVSupervisorIsBigEndian ()) {

```

Comment:
Why is this relevant? UEFI byte order is always little endian.

Reply:
RISC-V processors support both big-endian and little-endian byte orders. Is there a specification requirement that mandates RISC-V to operate in little-endian byte order?